populator-machinery/controller.go

```diff
@@ -108,13 +114,65 @@ type controller struct {
 	recorder             record.EventRecorder
 	referenceGrantLister referenceGrantv1beta1.ReferenceGrantLister
 	referenceGrantSynced cache.InformerSynced
+	useProviderImpl      bool
+	populate             func(context.Context, *PopulatorParams) error
+	populateComplete     func(context.Context, *PopulatorParams) (bool, error)
+}
+
+type VolumePopulatorConfig struct {
+	MasterURL     string
+	Kubeconfig    string
+	ImageName     string
+	HttpEndpoint  string
+	MetricsPath   string
+	Namespace     string
+	Prefix        string
+	Gk            schema.GroupKind
+	Gvr           schema.GroupVersionResource
+	MountPath     string
+	DevicePath    string
+	PopulatorArgs func(bool, *unstructured.Unstructured) ([]string, error)
+	// Boolean flag which determines whether or not to use cloud provider specific data populate functions
+	UseProviderImpl bool
+	// Data population function, invoked when the useProviderImpl variable is set to true
+	Populate func(context.Context, *PopulatorParams) error
```

Comment:
I would probably call this `PopulateFn` to make it clear this is a function pointer. Same for `PopulateComplete`

Reply:
Updated